evm/contracts/examples/ConversionRate.sol

```diff
@@ -177,16 +177,61 @@ contract ConversionRate is ChainlinkClient, Ownable {
     uint256 sumQuotients;
     uint256 sumRemainders;
     Answer memory answer = answers[_answerId];
-    uint256 responseCount = answer.responses.length;
-    for (uint i = 0; i < responseCount; i++) {
-      uint256 response = answer.responses[i];
-      sumQuotients = sumQuotients.add(response.div(responseCount)); // aggregate responses and protect from overflows
-      sumRemainders = sumRemainders.add(response % responseCount);
+
+    uint256 responseLength = answer.responses.length;
+    uint256 middleIndex = responseLength / 2;
+    if (responseLength % 2 == 0) {
+      uint256 median1 = answers[_answerId].responses[middleIndex];
+      uint256 median2 = answers[_answerId].responses[middleIndex - 1];
+      currentRate = median1.add(median2).div(2);
```

Comment:
A comment here, too, about the implicit security assumption that over 50% of the participants are honest, so the middle pair will not try to overflow the value. Or perhaps resorting to the remainders-and-quotients trick again.